### ISSUE TITLE
Incorrect error message: type parameter needs to be Sized, despite explicit bound

### ISSUE THREAD

**Issue by e2-71828** [rust-lang/rust#74149]
For the code below, the compiler complains that `Item` needs to be `Sized`, despite that being an explicitly-stated bound.

```rust
const PAGE_SIZE:usize=4096;

#[repr(align(4096))]
struct Page<Item: std::marker::Sized> {
    arr: [Item; PAGE_SIZE/std::mem::size_of::<Item>()],
}

```

([Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=87cb5e4f33cee354755117fda99ca20f))

Errors:

```
   Compiling playground v0.0.1 (/playground)
error[E0277]: the size for values of type `Item` cannot be known at compilation time
   --> src/lib.rs:5:47
    |
4   | struct Page<Item: std::marker::Sized> {
    |             ---- this type parameter needs to be `std::marker::Sized`
5   |     arr: [Item; PAGE_SIZE/std::mem::size_of::<Item>()],
    |                                               ^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `Item`
    = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `playground`.

To learn more, run the command again with --verbose.

```

**Comment by lcnr** [comment]
This is blocked on lazy normalization of constants, see #72219 for more details,
you can use `#![feature(lazy_normalization_consts)]` on nightly to try experiment
with the still incomplete implementation.

**Comment by lcnr** [comment]
Closing this in favor of #43408

**Comment by e2-71828** [comment]
Note that this bug isn’t about the ability to define this per se, but that the compiler is emitting an error that is nonsensical on its face— if this isn’t possible, the compiler should say so, insted of complaining about a falsehood.

**Comment by lcnr** [comment]
That's true, #43408 currently seems to track both the error message and the more general problem.
While this isn't optimal, considering how many people already subscribed to that issue I personally think it's better to keep it this way.
